File: w09/d01/enoch_riese/mail/lib/mailbox.rb
require_relative "package"
require_relative "postcard"

class Mailbox
  attr_accessor :objects
  def initialize
    @objects = []
  end
  def accept_mail(obj)
    @objects << obj
  end
  def open_mail
    @objects.pop
  end
  def to_s
    packs = 0
    posts = 0
    @objects.each do |mail|
      if mail.is_a?(Package)
        packs += 1
      elsif mail.is_a?(Postcard)
        posts += 1
      end
    end
    "The mailbox has #{posts} postcards and #{packs} packages"
  end
end

File: w09/d03/enoch_riese/mail_objects_app/app/assets/javascripts/mailbox.js
function Mailbox() {
  this.items = []
}

Mailbox.prototype.acceptMail = function(item) {
  this.items.push(item);
};

Mailbox.prototype.openMail = function() {
  last = this.items.pop();
  return last;
}

Mailbox.prototype.toString = function() {
  var packs = 0;
  var posts = 0;
  for (var i = this.items.length - 1; i >= 0; i--) {
    if (this.items[i] instanceof Package) {
      packs ++;
    }
    else if (this.items[i] instanceof Postcard) {
      posts ++;
    };
  };
  return "The Mailbox has " + posts + " postcards and " + packs + " packages"
}
